src/controllers/trainee/Controller.ts

```diff
@@ -13,33 +13,57 @@ class TraineeController {
             return TraineeController.instance;
 
       }
-      userRepository: UserRepository= new UserRepository();
-       
-      public get(req, res, next){
-            
-            
-                  let {limit = 0, skip = 0 } = req.query;
-                  skip= Number(skip);
-                  limit = Number(limit);
-
-             userRepository.getAll(limit, skip, {sort : {name: -1, email: -1}})
-                  .then((data) => {
-                     
-                        
-                        res.status(200).send({
-                              status: 'ok',
-                              message: 'Fetched Successfully',
-                              Trainees : data
-                        });
-                  })
-                  .catch((err) => {
-                        res.send({
-                              message: 'Not Fetched',
-                              status : 404
+      userRepository: UserRepository = new UserRepository();
+
+      public get(req, res, next) {
+
+            let { limit = 0, skip = 0, searchBy } = req.query;
+            skip = Number(skip);
+            limit = Number(limit);
+
+            const options = {
+                  limit,
+                  skip,
+                  sort: { name: -1, email: -1 },
+            }
+
+            let sort: any;
+            let trainee: any;
+            //let search: any;
+            const search = searchBy ? searchBy.toLowerCase() : "";
+            console.log("search value", search)
+            if (search) {
+                  function escapeRegExp(text) {
+                        return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, '\\$&');
+                  }
+                  const regex = new RegExp(escapeRegExp(req.query.searchBy), 'gi');
```

Comment:
remove this and add at the top these line
const query = {}
if(searchText) {
  query['$or'] = [{name: new ReExp(searchText, 'i')},{email:new ReExp(searchText, 'i')}]
}
pass query to the first argument of userRepository.getAll method